### PR TITLE
Add Hashflow trade tables.

### PR DESCRIPTION
**Pull request by gxmxni**
This PR needs to wait until the pool contract ABI is decoded on BSC and Polygon as well.

Tested queries as much as we could, but may need help tuning if things look funky.  wagmi

I've checked that:

* [x] the query produces the intended results
* [x] the folder name matches the schema name
* [x] the schema name exists in Dune
* [x] views are prefixed with `view_`, functions with `fn_`.
* [x] the filename matches the defined view, table or function and ends with .sql
* [x] each file has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

### CODE REVIEW

File: ethereum/dex/trades/insert_hashflow.sql [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        taker_token_amount * 10 ^ erc20b.decimals as token_b_amount_raw,
+        usd_amount,
+        maker_token as token_a_address,
+        taker_token as token_b_address,
+        router_contract as exchange_contract_address,
+        tx_hash,
+        tx."from" as tx_from,
+        tx."to" as tx_to,
+        NULL::integer[] as trace_address,
+        (case when compostie_index=-1 then NULL::integer else composite_index end) as evt_index, -- -1 means decoded from traces
```
Comment (masquot) [inline ethereum/dex/trades/insert_hashflow.sql:51]: composite_index?

Comment (masquot) [inline ethereum/dex/trades/insert_hashflow.sql:51]: composite_index ?

**Review by masquot — CHANGES_REQUESTED** [review]
Thanks so much @gxmxni for adding Hashflow.

I reviewed the files related to Ethereum. Overall, the PR looks excellent, but there are some typos and references to non existing columns.

Are you familiar with `dune_user_generated`? We recommend testing your queries there with a subset of the data e.g. setting `start_ts` to '2022-03-01' and `end_ts` to '2022-03-10'. More info on how to use `dune_user_generated` is here: https://docs.dune.xyz/data-tables/data-tables/user-generated

File: ethereum/hashflow/trades.sql [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+    select      evt_tx_hash as tx_hash,
+                evt_index,
+                trader,
+                txid as tx_id,
+                "quoteToken" as maker_token,
+                "baseToken" as taker_token,
+                "quoteTokenAmount" as maker_token_amount,
+                "baseTokenAmount" as taker_token_amount
+     from hashflow."Pool_evt_Trade0"
+          AND block_time >= start_ts AND block_time < end_ts
```
[the remark this answers is not in the document]
Comment (pani-poori) [reply ethereum/hashflow/trades.sql]: updated in new commits! 🙏

tested insert function: https://dune.xyz/queries/542690
test results looks good: https://dune.xyz/queries/542695

File: ethereum/hashflow/trades.sql [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+    select      evt_tx_hash as tx_hash,
+                evt_index,
+                trader,
+                txid as tx_id,
+                "quoteToken" as maker_token,
+                "baseToken" as taker_token,
+                "quoteTokenAmount" as maker_token_amount,
+                "baseTokenAmount" as taker_token_amount
+     from hashflow."Pool_evt_Trade"
+          AND block_time >= start_ts AND block_time < end_ts
```
[the remark this answers is not in the document]
Comment (pani-poori) [reply ethereum/hashflow/trades.sql]: fixed

[the remark this answers is not in the document]
Comment (pani-poori) [reply ethereum/hashflow/trades.sql]: fixed

File: ethereum/dex/trades/insert_hashflow.sql (hunk 1 above)
Comment (pani-poori) [reply ethereum/dex/trades/insert_hashflow.sql:51]: we decode trades from traces in the cases of:
- early records (legacy trades from last year alpha launch - when we didn't standardize the events)
- reverted trades (where there is no event fired)

thus there is no evt_index accompanied - we put a `-1` on those cases and when inserting into `dex.trades` -- we filter to only insert successful fills -- which will be a unique index.

Comment (pani-poori) [reply ethereum/dex/trades/insert_hashflow.sql:51]: same as above

**Comment by pani-poori** [comment]
also tested BSC and Polygon insert functions:


[BSC test](https://dune.xyz/queries/542706)
[BSC test result](https://dune.xyz/queries/542708)


[Polygon test](https://dune.xyz/queries/542697)
[Polygon test result](https://dune.xyz/queries/542704)

**Review by antonio-mendes — CHANGES_REQUESTED** [review]
Actually even though BSC and Polygon seem to work, Ethereum still has issues.
We'll provide a more detailed review during next week!

**Comment by gxmxni** [comment]
hi @antonio-mendes -- are there any new findings?

**Comment by antonio-mendes** [comment]
I went ahead and retested mainnet and it went through:
https://dune.com/queries/626588
https://dune.com/queries/626604

Will deploy this tomorrow, hopefully we don't run into any issues inserting into dex.trades this time.

File: polygon/dex/trades/insert_hashflow.sql [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        trade_id
+    )
+    SELECT
+        hf.block_time,
+        maker_symbol AS token_a_symbol,
+        taker_symbol AS token_b_symbol,
+        maker_token_amount AS token_a_amount,
+        taker_token_amount AS token_b_amount,
+        'hashflow' as project,
+        NULL as version,
```
Comment (antonio-mendes) [inline polygon/dex/trades/insert_hashflow.sql:37]: Here it should be `'1' as version`

File: polygon/dex/trades/insert_hashflow.sql [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        taker_token_amount * 10 ^ erc20b.decimals as token_b_amount_raw,
+        usd_amount,
+        maker_token as token_a_address,
+        taker_token as token_b_address,
+        router_contract as exchange_contract_address,
+        tx_hash,
+        tx."from" as tx_from,
+        tx."to" as tx_to,
+        NULL::integer[] as trace_address,
+        (case when compostie_index=-1 then NULL::integer else composite_index end) as evt_index, -- -1 means decoded from traces
```
Comment (antonio-mendes) [inline polygon/dex/trades/insert_hashflow.sql:51]: Typo on `compostie_index`, also should be `hf.composite_index` so it's clearer where it's coming from

File: polygon/dex/trades/insert_hashflow.sql [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    FROM hashflow.trades hf
+    INNER JOIN polygon.transactions tx
+            ON hf.tx_hash = tx.hash
+            AND tx.block_time >= start_ts
+            AND tx.block_time < end_ts
+            AND tx.block_number >= start_block
+            AND tx.block_number < end_block
+    LEFT JOIN erc20.tokens erc20a ON erc20a.contract_address = hf.maker_token
+    LEFT JOIN erc20.tokens erc20b ON erc20b.contract_address = hf.taker_token
+    WHERE tx_status is true -- success trade
```
Comment (antonio-mendes) [inline polygon/dex/trades/insert_hashflow.sql:62]: This variable doesn't exist on hf, it's actually called `fill_status`

**Review by antonio-mendes — COMMENTED** [review]
Few changes that need to be done but even then this is failing to insert to `dex.trades`.

Error is:
```
LINE 48: ...oject, tx_hash, evt_index, trace_address ORDER BY version, c...
                                                              ^
HINT:  There is a column named "version" in table "trades", but it cannot be referenced from this part of the query.
```

**Comment by jeff-dude** [comment]
@gxmxni as antonio mentioned, there are a few issues loading dex trades from the hashflow trades table. we don't have access to push changes to your forked repo. i'll include the inner select statement from polygon/dex/trades/insert_hashflow.sql script (lines 30-64), with fixes, that you can use as a guide to make the necessary changes (or grant us access to do so).

https://dune.com/queries/650520

**Comment by antonio-mendes** [comment]
Successfully deployed on Ethereum and Polygon 🎉 
Screenshot 2022-05-05 at 17 10 45

**Comment by jeff-dude** [comment]
new PR #1036 created to handle this request

**Pull request closed without merging**